_overviews/overview/a-taste-of-scala.md

```diff
@@ -0,0 +1,1329 @@
+
+Here are three traits that define well-organized and modular behaviors for animals like dogs and cats:
+
+```scala
+trait Speaker:
+  def speak(): String  // has no body, so it’s abstract
+
+trait TailWagger:
+  def startTail(): Unit = println("tail is wagging")
+  def stopTail(): Unit = println("tail is stopped")
+
+trait Runner:
+  def startRunning(): Unit = println("I’m running")
+  def stopRunning(): Unit = println("Stopped running")
+```
+
+Given those traits, here’s a `Dog` class that extends all of those traits while providing a behavior for the abstract `speak` method:
+
+```scala
+class Dog(name: String) extends Speaker with TailWagger with Runner:
+  def speak(): String = "Woof!"
+```
+
+Notice how the class extends the traits with the `extends` and `with` keywords.
+
+Similarly, here’s a `Cat` class that implements those same traits while also overriding two of the concrete methods it inherits:
+
+```scala
+class Cat(name: String) extends Speaker with TailWagger with Runner:
+  def speak(): String = "Meow"
+  override def startRunning(): Unit = println("Yeah ... I don’t run")
+  override def stopRunning(): Unit = println("No need to stop")
+```
+
+These examples show how those classes are used:
+
+```scala
+val d = Dog("Rover")
+d.speak()               // prints "Woof!"
+
+val c = Cat("Morris")
+c.speak()               // "Meow"
+c.startRunning()        // "Yeah ... I don’t run"
+c.stopRunning()         // "No need to stop"
+```
+
+<!-- TODO: links  -->
+If that code makes sense — great, you’re comfortable with traits as interfaces. If not, don’t worry, they’re explained in more detail in the Data Modeling sections of this Overview and the Reference documentation.
+
+<!-- TODO: show traits that take parameters here? -->
+
+
+#### Classes
+
+<!--
+- TODO: update for final/open/closed
+- https://dotty.epfl.ch/docs/reference/other-new-features/open-classes.html
+- An `open` modifier on a class signals that the class is planned for extensions
+- Classes that are not open can still be extended, but only if at least one of two alternative conditions is met:
+  - The extending class is in the same source file as the extended class
+  - The language feature adhocExtensions is enabled for the extending class
+- Details
+  - `open` is a soft modifier. It is treated as a normal identifier unless it is in modifier position.
+  - An open class cannot be final or sealed.
+  - Traits or abstract classes are always open, so open is redundant for them.
+-->
+
+Scala *classes* are used in OOP-style programming. Here’s an example of a class that models a “person.” The first and last names can both be modified, so they’re declared as `var` parameters:
+
+```scala
+class Person(var firstName: String, var lastName: String):
+  def printFullName() = println(s"$firstName $lastName")
+
+val p = Person("Julia", "Kern")
+println(p.firstName)   // "Julia"
+p.lastName = "Manes"
+p.printFullName()      // "Julia Manes"
+```
+
+
+### FP data modeling 
```

Comment:
I ithink it would be nice to write “functional programming” rather than “FP”. What do you think of the following?

```suggestion
### Data Modeling (Functional Programming Style)
```

Reply:
I like it!